src/obj_dir.ml

```diff
@@ -1,127 +1,241 @@
 open! Stdune
 open Import
 
+module External = struct
+  type t =
+    { public_dir : Path.t
+    ; private_dir : Path.t option
+    }
+
+  let make ~dir ~has_private_modules =
+    let private_dir =
+      if has_private_modules then
+        Some (Path.relative dir ".private")
+      else
+        None
+    in
+    { public_dir = dir
+    ; private_dir
+    }
+
+  let has_public_cmi_dir t = Option.is_some t.private_dir
```

Comment:
Shouldn't this be `has_private_cmi_dir`?

Reply:
Not quite. AFAIK, when @bobot wrote this, this check is true whenever we have a dedicated dir public cmi's. In the case where we don't have a private_dir, there is no _dedicated_ dir for public cmi's. But in fact, this check is only necessary when compiling stuff. So perhaps we can even error out when this is called on an external dir, as it doesn't mean all that much anymore.

Reply:
What bothers me is that the function is called `has_public`  and in fact inspect a field called `private`. That looks weird. If there is a good reason for writing the code like this, then they should be a comment explaining it.

Reply:
I'm not happy about the name either, should I change it here?

Essentially the meaning is as follows: The library has private modules hence the public cmi's need to live in a dedicated dir. Any suggestion for a better name?

Reply:
Yh, might as well do that now. I suggest `need_dedicated_public_dir`. It seems more connected to the meaning.